### DISCUSSION TITLE
Project Update May 2023

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#148] [Announcements]
Well, we are well into May already and thought it time to write an update. 

There are no real code changes to the core code. The current Master and Release still represent the latest release candidate. I'm reluctant to undertake further development on any of the V3 features until the current version has been finalised. The reason for this is that development on a new branch might stray too far from the current codebase and result in scope creep, which this project has already been guilty of a number of times. So what is really needed at this stage is further uptake and testing so that we can iron out any issues with the V2 code.

### Air Bleed Control

On a somewhat related note I finally received the DBW throttle body that I ordered (twice). The initial vendor sent me the wrong part. After wrangling with them for weeks trying to get the issue resolved, I ended up ordering another part from a different vendor.

![PXL_20230509_100222308 MP](https://github.com/DeeEmm/DIY-Flow-Bench/assets/3038710/b565843f-17ad-454a-9aed-e076ebca3605)

I also dug out the PWM controller that I have for another project to test with.

![PXL_20230509_100236851](https://github.com/DeeEmm/DIY-Flow-Bench/assets/3038710/fe89afac-a81f-4767-bddb-08e88c04f88f)

Theoretically this can be used to drive the throttle body and if successful will allow us to integrate the controller with the shield so that it can be used as an air bleed control. Integration is fairly trivial as from memory the driver just needs an analog reference. 

### Flow Measurement Fixture

I have also received the load cells and load cell amplifier that I ordered to set up an orifice flow test jig. If you've not been reading the discussion on this, the idea is to measure the volume (by weight) of distilled water than an orifice will flow so that it can be converted to air volume. If successful this will allow any orifice to be accurately measured without needing access to an existing  flow bench. It also potentially addresses the inherent errors present in most flow benches by removing the reliance on an existing orifice to determine flow values. 

![PXL_20230510_054918161](https://github.com/DeeEmm/DIY-Flow-Bench/assets/3038710/5314567c-740a-49b1-94cf-cdcffe10ae8d)

The load cell will then be used to get started on the [DIY Dyno project](https://github.com/DeeEmm/DIY-Dyno) which will be largely ported from the DIYFB code.

### Please Help. Contributions Needed

If you are considering building a flow bench, please consider using the DIYFB code and help contribute to the project by providing feedback and measurement results. It is really important to get a variety of results so that we can compare a range of different benches and gauge accuracy and repeatability. 

At this stage in the project we really need to validate that what we have works and works reliably and accurately. There is little value in continuing until we can be confident that both the code and hardware generates accurate repeatable results.

/DM

**Comment by black-top** [comment]
I didn't notice a new topics. A lot of progress is made.

**Reply by DeeEmm** [reply]
haha, not so much progress, just retail therapy lol
